### ISSUE TITLE
Some mobs are always taking two swings

### ISSUE THREAD

**Issue by Hozu** [DarkstarProject/darkstar#2803]
Client Version (type /ver in game) :
30160203_0

Server Version (type @revision in game) :
3a852af

Source Branch (master/stable) :
master

Additional Information (Steps to reproduce/Expected behavior) :
The mobs in question are seemingly random. Or at least, I can't discern a pattern. So far I've identified Veteran Quadav, Yagudo Persecutor, and Marquis Andras. None of them should be swinging twice per round due to their job and/or level, let alone every time.

**Comment by RwNigma** [comment]
I believe MNK type mobs have this problem as well. The Gigas in Upper Delkfutt Tower. Gatekeeper I believe. Looking at the weapon delay of 240 within the database it may work differently with H2H weapons at this delay.

**Comment by Hozu** [comment]
MNK mobs and delay? Oh... oh no....

I may have introduced a bug with that... are MNK mobs considered having h2h skill?

**Comment by RwNigma** [comment]
To be honest I am not sure. I just know that they swing at a speed slightly less than hundred fists. If the mobs you've listed are swinging with double strike then think quadrastrike for certain/all MNK mobs and that's how fast they swing.

**Comment by Hozu** [comment]
I don't think they are, but they have the double swing property for being MNK. Since the delay check is checking h2h skill... yeah. For mobs it needs to check h2h damage type, players skill (because of Birdbanes/Vampiric Claws). I'll have to fix that one in a little bit.

**Comment by Hozu** [comment]
Well I can't figure out why it's doing that anymore. Damn.

**Comment by Deadwing888** [comment]
This is a problem with Seiryu as well.

**Comment by teschnei** [comment]
I did a bit of digging, it's because the mobs weapon skill type is hand-to-hand, which comes from the cmbSkill column in the database.  Currently not known why mobs even have a combat skill, as they can't equip any weapons.

**Comment by Hozu** [comment]
Maybe it was for MNK mobs that aren't supposed to swing tw- ... nah I got nothing.

**Comment by Hozu** [comment]
hm since this wasn't an issue before the ai rewrite, maybe that column was used for something else?

**Comment by Hozu** [comment]
Maat is swinging stupidly fast on all jobs with this issue, with two swings regardless of job.

**Comment by Hozu** [comment]
Wait, I think the cmbSkill column would be the mob's skill rank, rather than their skill type, though that means for non-players it should probably be checking if the mob is a MNK instead of skill... though some MNK mobs don't have double swing (seiryu, maybe the MNK demon NM in xarc?), and some non-MNK mobs do have it (NIN aerns).

**Comment by teschnei** [comment]
what use would a skill rank column be for mobs?

**Comment by Hozu** [comment]
Good point, about as useful as a skill type column I suppose, heh.

**Comment by Hozu** [comment]
I haven't been able to come up with any solution for this one, but people are having an extremely hard time beating Maat because of this.

**Comment by teschnei** [comment]
isn't he supposed to attack twice?

**Comment by Hozu** [comment]
IIRC only on jobs that have native h2h skill.

**Comment by waytim** [comment]
So after researching a bit, mobs like Alkyoneus that are MNK as main or subjob (he's war/mnk) and using cmbskill 1 (h2h) have crazy high dbl/triple attack proc rates, almost 100%. Charybdis also has stopped attacking 6 times which leads me to believe it's this area and below that's causing the issues:
https://github.com/DarkstarProject/darkstar/blob/314c2a1fc93d68c89b47984bf0b29187101e31aa/src/map/utils/battleutils.cpp#L2516

**Comment by abriasffxi** [comment]
I agree- will be testing removing cmbskil 1 for mobs with sub mnk later today.  It happens with many of the beastman mobs (Orcish Grunts, Sentinels, and the associated Quadavs, haven't checked Yagudo).

**Comment by Hozu** [comment]
IMO cmbskill should be replaced by isH2H, and have the field be either 0 or 1. However, I bet the skill is also used to calculate the attack/acc for the mob based on their job.

**Comment by abriasffxi** [comment]
All - I was successful in changing all of my known instances of this bug with one query (from memory so please double check):

UPDATE dspdb.mob_pools SET cmbSkill=0 WHERE (cmbSkill=1 AND mJob <> 2 AND sJob <>2);

Not sure if mJob and sJob are exactly correct but this had about ~200mobs in the database that are not identified as monks, either by main or sub job, that it fixed.  I'm sure there are some thf mobs using h2h that it messed up but I didn't go through and fix them individually.  I feel like if someone checks that query and commits each individually if they know the mobs well it will be a pretty strong fix.

**Comment by HeavensSword** [comment]
I would advise against a massive UPDATE like that as some mobs still need to attack twice even without mnk main or sub.

For instance:
I'm in the middle of a change for mandragoras and some of the NM are rdm/blm, but still need to attack twice.

**Comment by abriasffxi** [comment]
Right, that's why I suggest someone runs the query and check the mobs.

**Comment by phlare** [comment]
here's a pastebin with the results of `SELECT poolid, name, familyid, mJob, sJob, cmbSkill, cmbDelay, mobType FROM mob_pools WHERE cmbSkill =1 AND mJob <> 2 AND sJob <> 2;`   just for perusing.

 We could create either a whitelist or blacklist of poolids for an `IN` clause and add that to the query above for a quick-and-dirty-but-not-so-destructive fix.

http://pastebin.com/Q3w5yqKw

**Comment by davismj** [comment]
Cold Gigas in Beaucedine. Flagged as Warrior.
